pkg-py/tests/test_querychat_new_api.py

```diff
@@ -0,0 +1,112 @@
+"""Tests for the new QueryChat and QueryChatCore API."""
```

Comment:
nit: We can fully commit to this approach, which means it doesn't have to be `querychat_new_api` and can just be The API.